fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/PhysicalTableConfiguration.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.provider;
+
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+
+import java.util.Set;
+
+/**
+ * A PhysicalTableConfiguration can create its PhysicalTableDefinition, given a dimension configuration.
+ */
+public interface PhysicalTableConfiguration {
+
```

Comment:
Hmm, some of the other classes are like this, too (e.g. the MetricConfiguration interface has a `build` method that returns a logical metric).

For physical tables, it would be straightforward to clear this up by exposing dimensions and remove this method. The `PhysicalTableDefinition` would be created by the consumer of this class (e.g. `ConfiguredTableLoader` or something like that).

For metrics, it's a little trickier. I didn't want to force use of the metric parser so I wanted implementers of `MetricConfiguration` to be able to specify how a metric is built. I guess I could split that out into a MetricConfiguration and some sort of MetricFactory.